sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/exception/CosmosBadRequestException.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos.exception;
+
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.BadRequestException;
+
+/**
+ * While this class is public, but it is not part of our published public APIs.
+ * This is meant to be internally used only by our sdk.
+ */
+public class CosmosBadRequestException extends BadRequestException {
+
+    /**
+     * Cosmos exception.
+     */
+    protected final CosmosException cosmosException;
```

Comment:
@kushagraThapar  this is what I was unsure about adding this but it does help minimize the breaking change and it preserves the inner exception. If, for example, you do CosmosBadRequest.getMessage() it is different than CosmosBadRequest.getCosmosException().getMessage(). This was breaking some of our exsiting tests as the expected message is now in the surpressedExceptions of the CosmosBadRequest. This will help minimize the level of the breaking change as the user can still get out the message they always have if they were using this format.

Reply:
@trande4884 I am curious, why don't we extend CosmosAccessException for all of these exceptions, so that it is not a breaking change. I don't think we will be able to do a breaking change in our spring data cosmos SDK without going for a major version change, which we are not planning right now. 

If all of these new exceptions extend CosmosAccessException, then it will not be a breaking change.